### DISCUSSION TITLE
Why would these errors occur?

### DISCUSSION THREAD

**Discussion by melmelchew** [xyflow/xyflow#3170] [Q&A]
Hello! I've been encountering a few errors that are not mentioned in the troubleshoting reactflow page or issues and discussion github pages and am hoping to see if others have also faced such runtime errors while using reactflow to find out why these errors are occurring. 
I am using reactflow v11.7.4 for a react typescript webapp.

The first error I would occasionally get is a runtime error as seen in the image:
<img width="500" alt="Screenshot 2023-06-23 at 2 13 26 PM" src="https://github.com/wbkd/react-flow/assets/83665067/354229a2-517b-4197-87bd-48c891772d5f">
This error occurs when I am dragging a group node (consequently moving the nodes linked to the group node through the `parentNode` property as well) around. Sometimes this error will happen, sometimes not. I am wondering if it has got to do anything with the way I have coded my app, and which reactflow prop function is using `getNodePositionWithOrigin` (maybe passing in some functions as prop into the Reactflow component could be the cause?) . 

The next error I would occasionally get is this other runtime error `ResizeObserver loop limit exceeded` :
<img width="500" alt="Screenshot 2023-06-23 at 2 57 39 PM" src="https://github.com/wbkd/react-flow/assets/83665067/58cc6136-addc-4b02-b8ab-18afbcac3598">
This happens when I am enlarging or shrinking the size of the node. I am not sure if it's because of certain prop specifications needed, will attach the code here:

```
import { FC, memo} from 'react';
import { Handle, Position, NodeProps, useReactFlow, useStoreApi, NodeResizer } from 'reactflow';
import {Typography } from '@mui/material';

const ClusterNode: FC<NodeProps> = ({ id, data, isConnectable, selected }) => {
    const { setNodes } = useReactFlow();
    const store = useStoreApi();

  return (
    <div className="cluster-node">
    <NodeResizer 
    minWidth={data.minWidth ?? undefined}
        maxWidth={data.maxWidth ?? undefined}
        minHeight={data.minHeight ?? undefined}
        maxHeight={data.maxHeight ?? undefined}
        isVisible={data.isVisible ?? selected}
        shouldResize={data.shouldResize ?? undefined}
        onResizeStart={data.onResizeStart ?? undefined}
        onResize={data.onResize ?? undefined}
        onResizeEnd={data.onResizeEnd ?? undefined}
        keepAspectRatio={data.keepAspectRatio ?? undefined} />
      <Handle type="target" position={Position.Top} isConnectable={isConnectable} />
      <div className='title'>
      <Typography variant="subtitle2">
      {data.label}
      </Typography>
      </div>
      
      <Handle type="source" position={Position.Bottom} id="b" isConnectable={isConnectable} />
    </div>
  );
}

export default memo(ClusterNode);
```
I have tried to test out the template code provided for node resizer [here](https://github.com/wbkd/react-flow/tree/main/examples/vite-app/src/examples/NodeResizer) but the output seems wrong 😅 (I have specified the node types and passed the prop into the Reactflow component, but somehow it is not detecting and just falling back to default nodes)
<img width="500" alt="Screenshot 2023-06-23 at 3 10 30 PM" src="https://github.com/wbkd/react-flow/assets/83665067/9d635a7b-8ca8-42a9-94e4-587775b364d2">

Thank you in advance to anyone for helping! Do let me know if there is anymore code or context I need to provide :)

**Comment by moklick** [comment]
Hey @melmelchew 

it's hard to tell if we can't see any code. I could imagine that you are accessing the nodes in a component which is not the flow itself. This often leads to performance issues since the component gets re-rendered every time you drag/select nodes. Do you have a repo or codesandbox we could check out?

**Reply by melmelchew** [reply]
Yep I have created a public repo [here](https://github.com/melmelchew/Flow)!

The call stack error can be usually triggered when I drag around parent nodes a lot or frequently, otherwise it may suddenly occur out of nowhere when I drag a child node.

The ResizeObserver error sometimes happens when I am expanding and reducing the size of the group node continuously (is a custom node called a ClusterNode).

Thank you very much for taking the time to look at my code!

**Reply by moklick** [reply]
I can't reproduce the issues you mentioned. Which browser are you using?

**Reply by melmelchew** [reply]
Chrome Version 114.0.5735.133

The errors do not usually occur, sometimes it will only be thrown after running the code for awhile if I were to aggressively drag the node around or resize it. 

I have been looking at my code to see if I am accessing the nodes in a component which is not in the flow itself, but so far I can't really find any telltale signs of that occurring. (then again, I am also unsure as to what I should be looking out for)

**Reply by moklick** [reply]
I am still unable to repro this. Could you try to remove `nodes` from the dep array of your `handleDragEnd` function and receive inside the handler via `getNodes` from `useReactFlow` ?

**Reply by melmelchew** [reply]
Did it and it works now! There is no more call stack error :) I guess by getting the list of nodes on the app level and not through the reactFlowInstance must have been causing the re-rendering. 

However the ResizeObserver loop limit error still occurs when I continuously resize the clusterNode that is generated from clicking the 'add group node' button. 

I have added some components that have caused the resize error to occur more frequently and have pushed these changes to the github. The steps I did to replicate the error is as follows:

1) upon starting up the app or refreshing the browser, I click on 'add group node' button.
2) I click on the newly appeared clusterNode (may have to move it away from the other initial nodes) to cause a sidebar to popup.
3) I drag the corner of the clusterNode while the sidebar is still popped up and resize the node.
4) The error should appear.
